vars/withLibraryPipeline.groovy

```diff
@@ -0,0 +1,190 @@
+import uk.gov.hmcts.contino.AngularPipelineType
+import uk.gov.hmcts.contino.azure.Acr
+import uk.gov.hmcts.contino.Builder
+import uk.gov.hmcts.contino.DockerImage
+import uk.gov.hmcts.contino.Environment
+import uk.gov.hmcts.contino.MetricsPublisher
+import uk.gov.hmcts.contino.NodePipelineType
+import uk.gov.hmcts.contino.PipelineType
+import uk.gov.hmcts.contino.ProjectBranch
+import uk.gov.hmcts.contino.SpringBootPipelineType
+import uk.gov.hmcts.contino.Subscription
+import uk.gov.hmcts.contino.AppPipelineConfig
+import uk.gov.hmcts.contino.AppPipelineDsl
+import uk.gov.hmcts.contino.PipelineCallbacksConfig
+import uk.gov.hmcts.contino.PipelineCallbacksRunner
+import uk.gov.hmcts.pipeline.AKSSubscriptions
+import uk.gov.hmcts.pipeline.deprecation.WarningCollector
+import uk.gov.hmcts.pipeline.TeamConfig
+
+def call(type, String product, String component, Closure body) {
+
+  def branch = new ProjectBranch(env.BRANCH_NAME)
+
+  def deploymentNamespace = branch.deploymentNamespace()
+  def deploymentProduct = deploymentNamespace ? "$deploymentNamespace-$product" : product
+
+  def pipelineTypes = [
+    java  : new SpringBootPipelineType(this, deploymentProduct, component),
+    nodejs: new NodePipelineType(this, deploymentProduct, component),
+    angular: new AngularPipelineType(this, deploymentProduct, component)
+  ]
+
+  Subscription subscription = new Subscription(env)
+  AKSSubscriptions aksSubscriptions = new AKSSubscriptions(this)
+
+  PipelineType pipelineType
+
+  if (type instanceof PipelineType) {
+    pipelineType = type
+  } else {
+    pipelineType = pipelineTypes.get(type)
+  }
+
+  assert pipelineType != null
+
+  MetricsPublisher metricsPublisher = new MetricsPublisher(this, currentBuild, product, component, subscription.prodName )
+  def pipelineConfig = new AppPipelineConfig()
+  def callbacks = new PipelineCallbacksConfig()
+  def callbacksRunner = new PipelineCallbacksRunner(callbacks)
+
+  callbacks.registerAfterAll { stage ->
+    metricsPublisher.publish(stage)
+  }
+
+  def dsl = new AppPipelineDsl(this, callbacks, pipelineConfig)
+  body.delegate = dsl
+  body.call() // register pipeline config
+
+  dsl.onStageFailure() {
+    currentBuild.result = "FAILURE"
+  }
+
+  Environment environment = new Environment(env)
+
+  node {
+    def slackChannel = new TeamConfig(this).getBuildNoticesSlackChannel(product)
+    try {
+      env.PATH = "$env.PATH:/usr/local/bin"
+
+
+
+      onPR {
+      }
+
+      onMaster {
```

Comment:
is there a reason this is only run on master?

Reply:
Yes we only want to publish contract tests on a master build. There's only 1 Pact broker currently.